l3build/l3build.lua

```diff
@@ -565,27 +565,13 @@ function mkdir(dir)
   end
 end
 
--- Find the relationship between two directories
-function relpath(target, source)
-  -- A shortcut for the case where the two are the same
-  if target == source then
-    return ""
-  end
-  local resultdir = ""
-  local trimpattern = "^[^/]*/"
-  -- Trim off identical leading directories
-  while
-    (match(target, trimpattern) or "X") ==
-    (match(target, trimpattern) or "Y") do
-    target = gsub(target, trimpattern, "")
-    source = gsub(source, trimpattern, "")
-  end
-  -- Go up from the source
-  for i = 0, select(2, gsub(target, "/", "")) do
-    resultdir = resultdir .. "../"
-  end
-  -- Return the relative part plus the unique part of the target
-  return resultdir .. target
+-- Return an absolute path from a relative one
```

Comment:
As noted in the first commit comment, the previous `relpath` was broken and I needed a working base to implement globbing on. Unfortunately, lua's limited filesystem functions lack the tools to properly implement a more elegant version, so I quickly hacked together `abspath` sans a better alternative.

As it temporarily changes the current working directory of the whole program, it should only be used with caution. I strongly suggest to replace it with something better in the future. Teaching the old `relpath` how to handle `.` and `..` could be the far superior alternative. A conversion function for MakeIndex is then easy to write.